cpp/src/arrow/type.h

```diff
@@ -439,7 +439,15 @@ struct ARROW_EXPORT DateType : public FixedWidthType {
   static std::string name() { return "date"; }
 };
 
-enum class TimeUnit : char { SECOND = 0, MILLI = 1, MICRO = 2, NANO = 3 };
+enum class TimeUnit : char {
+  SECOND = 0,
+  MILLI = 1,
+  MICRO = 2,
+  NANO = 3,
+
+  MIN = SECOND,
+  MAX = NANO,
```

Comment:
`TimeUnit` is a strongly typed enum, so these MIN/MAX fields aren't necessary. In Flatbuffers, the MIN/MAX values are these to help account for NULL.